src/main/java/com/couchbase/lite/javascript/ViewReduceBlockRhino.java

```diff
@@ -58,36 +66,37 @@ public ViewReduceBlockRhino(String src) {
 
     @Override
     public Object reduce(List<Object> keys, List<Object> values, boolean reReduce) {
-
         switch (nativeReduce) {
             case SUM:
                 return nativeSum(keys, values, reReduce);
             case COUNT:
                 return nativeCount(keys, values, reReduce);
             case DEFAULT:
             default:
-                org.mozilla.javascript.Context ctx = org.mozilla.javascript.Context.enter();
-                try {
-                    ctx.setOptimizationLevel(-1);
-                    ctx.setWrapFactory(wrapFactory);
-
-                    Scriptable localScope = ctx.newObject(globalScope);
-                    localScope.setPrototype(globalScope);
-                    localScope.setParentScope(null);
-
-                    Object[] args = new Object[3];
-
-                    args[0] = org.mozilla.javascript.Context.javaToJS(keys, localScope);
-                    args[1] = org.mozilla.javascript.Context.javaToJS(values, localScope);
-                    args[2] = org.mozilla.javascript.Context.javaToJS(reReduce, localScope);
-
-                    return reduceFunction.call(ctx, localScope, null, args);
-
-                } catch (org.mozilla.javascript.RhinoException e) {
-                    // TODO check couchdb behaviour on error in reduce function
-                    return null;
-                } finally {
-                    org.mozilla.javascript.Context.exit();
+                synchronized (lockFunction) {
```

Comment:
I don't see any threading issue with `nativeSum()` and `nativeCount()`. These methods don't access any instance variables.